msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -7728,12 +7729,17 @@ private DiscussionTopicBean duplicateTopic(Long originalTopicId, DiscussionForum
 			log.debug("About to get a member of membershipIter");
 			DBMembershipItem oldItem = (DBMembershipItem)membershipIter.next();
 				log.debug("About to getMembershipItemCopy()");
-				DBMembershipItem newItem = getMembershipItemCopy(oldItem);
-				if (newItem != null) {
-					newItem = permissionLevelManager.saveDBMembershipItem(newItem);
-					newTopic.addMembershipItem(newItem);
+				DBMembershipItem itemCopy = getMembershipItemCopy(oldItem);
+				if (itemCopy != null) {
+					DBMembershipItem newItem = permissionLevelManager.saveDBMembershipItem(itemCopy);
+					membershipItemSet.add(newItem);
 				}
 		}
+
+		// This will bond the new permission items to the topic with surrogate key
+		newTopic.setMembershipItemSet(membershipItemSet);
+		DiscussionTopic finalNewTopic = newTopic;
+		membershipItemSet.forEach(i -> ((DBMembershipItemImpl) i).setTopic(finalNewTopic));
```

Comment:
It looks like the issue is that the newItem created on 7734 was never being updates with the new topic?
This likely could be solved with a single line above when you create the newItem just call newItem.setTopic(newTopic).